src/utils/work/seekWorkResponse.js

```diff
@@ -61,11 +61,14 @@ const dispatchWorkRequest = async (
 
   const responsePromise = new Promise((resolve, reject) => {
     io.on(`WorkResponse-${ETag}`, (res) => {
-      const { response: { error } } = res;
+      const { response } = res;
+
+      if (response?.error) {
+        const { errorCode, userMessage } = response;
+        console.error(errorCode, userMessage);
 
-      if (error) {
         return reject(
-          new WorkResponseError(error, request),
+          new WorkResponseError(errorCode, userMessage, request),
```

Comment:
will there always be both `userMessage` and `errorCode` in the response when there is an error?

Reply:
Yes, there will always be a `userMessage` and `errorCode` when there is an error. The error response is built in `tasks.py` : https://github.com/hms-dbmi-cellenics/worker/blob/master/python/src/worker/tasks/factory.py